File: vane-proxy-core/src/main/java/org/oddlama/vane/proxycore/commands/ProxyCommand.java
package org.oddlama.vane.proxycore.commands;

import java.util.UUID;
import org.oddlama.vane.proxycore.VaneProxyPlugin;

public abstract class ProxyCommand {

    public final String permission;
    public final VaneProxyPlugin plugin;

    public ProxyCommand(String permission, VaneProxyPlugin plugin) {
        this.permission = permission;
        this.plugin = plugin;
    }

    public abstract void execute(ProxyCommandSender sender, String[] args);

    @SuppressWarnings("BooleanMethodIsAlwaysInverted")
    public boolean hasPermission(UUID uuid) {
        return this.permission == null || plugin.getProxy().hasPermission(uuid, this.permission);
    }
}

File: vane-proxy-core/src/main/kotlin/org/oddlama/vane/proxycore/commands/ProxyCommand.kt
package org.oddlama.vane.proxycore.commands

import org.oddlama.vane.proxycore.VaneProxyPlugin
import java.util.*

abstract class ProxyCommand(val permission: String?, @JvmField val plugin: VaneProxyPlugin) {
    abstract fun execute(sender: ProxyCommandSender?, args: Array<String?>?)

    fun hasPermission(uuid: UUID?): Boolean {
        return this.permission == null || plugin.proxy.hasPermission(uuid, this.permission)
    }
}
